XiEditor/Storage/LanguagesStore.swift

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+protocol LanguagesProvider {
+    func fetchRecentlyUsedLanguages() -> [String]
+    func saveRecentlyUsed(language: String)
+    var availableLanguages: [String] { get set }
+}
+
+class LanguagesStore: LanguagesProvider {
+    
+    enum Constants {
+        static let MaxRecentlyUsedLanguages = 5
+    }
+
+    private enum Keys: String {
+        case recentlyUsedLanguages
+    }
+
+    private let keyValueStorage: KeyValueStorage
+    var availableLanguages: [String] = []
+
+    public init(withStorage keyValueStorage: KeyValueStorage) {
+        self.keyValueStorage = keyValueStorage
+    }
+
+    func fetchRecentlyUsedLanguages() -> [String] {
+        return keyValueStorage.stringArray(forKey: Keys.recentlyUsedLanguages.rawValue) ?? []
+    }
+
+    func saveRecentlyUsed(language: String) {
+        
+        let newLanguage = language.trimmingCharacters(in: .whitespacesAndNewlines)
+        
+        if newLanguage.count == 0 {
```

Comment:
This check seems like a perfect job for a `guard` 😄 
`guard newLanguage.count > 0 else {
    return
}`

Reply:
@mmatoszko you think `if` can't handle it? 😎 It is not a big difference in this situation what to use. I think it easy to read simple `if` than `guard`. `guard` is good when you need to declare variables but do not want to increase cyclomatic complexity. Yes, compiler checks and if there is no `return` at some point it breaks compilation, but in such case (having tests) it is not so critical.

Reply:
totally agree :+1: